Shared/AddLocalDataController.cpp

```diff
@@ -808,15 +809,47 @@ void AddLocalDataController::createTiledLayer(const QString& path, int layerInde
     if (operationalLayers)
       operationalLayers->append(vectorTiledLayer);
 
-    emit layerSelected(vectorTiledLayer);
-    Q_UNUSED(layerIndex)
+    emit layerSelected(vectorTiledLayer);    
   }
   else
   {
     emit layerCreated(layerIndex, vectorTiledLayer);
   }
 }
 
+/*!
+ \brief Creates a KmlLayer with the provided KML path.
+
+ \list
+   \li \a path - The path to the local data source.
+   \li \a layerIndex - The index for which the layer will be added to the operational layer list.
+   \li \a visible - Whether the layer should be visible by default.
+   \li \a autoAdd - Whether the layer will be automatically added to the operational layer list.
+        If \c false, it will not add automatically. Instead, a signal will emit once the Layer has
+        been constructed.
+ \endlist
+*/void AddLocalDataController::createKmlLayer(const QString& path, int layerIndex, bool visible, bool autoAdd)
+{
+  KmlDataset* kmlDataset = new KmlDataset(path, this);
```

Comment:
I missed this on the first pass, but this should use the proper conversion for QUrl and local files.

```suggestion
  KmlDataset* kmlDataset = new KmlDataset(QUrl::fromLocalFile(path), this);
```